### PR TITLE
feat(integrations): Migrate Jira plugins to Jira integration

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
Adds a button to the Jira Cloud and Jira Server configuration pages that when clicked, migrates the plugin linked issues and ignored fields and disables the plugin(s). This should only appear if there are plugins to migrate.

Related Backend PR: https://github.com/getsentry/sentry/pull/37577

## UI:
jira plugin migration ui

### CODE REVIEW

File: static/app/views/settings/organizationIntegrations/configureIntegration.tsx [hunk 1]
```diff
@@ -147,6 +189,41 @@ class ConfigureIntegration extends AsyncView<Props, State> {
...
+              renderMessage={() => (
+                <Fragment>
+                  <p>
+                    {t(
+                      'This will automatically associate all the Linked Issues of your Jira Plugins to this integration.'
+                    )}
+                  </p>
+                  <p>
+                    {t(
+                      'If the Jira Plugins had the option checked to automatically create a Jira ticket for every new Sentry issue checked, you will need to create alert rules to recreate this behavior.'
```
Comment (ceorourke) [inline static/app/views/settings/organizationIntegrations/configureIntegration.tsx:207]: Jira Server unfortunately doesn't have the automatic ticket creation alert rule option :/ Can you call that out?

File: static/app/views/settings/organizationIntegrations/configureIntegration.tsx [hunk 2]
```diff
@@ -126,8 +140,36 @@ class ConfigureIntegration extends AsyncView<Props, State> {
...
+    } catch (error) {
+      addErrorMessage(t('Something went wrong! Please try again.'));
+    }
+  };
   getAction = (provider: IntegrationProvider | undefined) => {
-    const {integration} = this.state;
+    const {integration, plugins} = this.state;
+    const shouldMigrateJiraPlugin =
+      provider &&
+      provider.key === 'jira' &&
```
Comment (ceorourke) [inline static/app/views/settings/organizationIntegrations/configureIntegration.tsx:170]: Will this work for Jira Server too or should it also check for the `jira_server` provider key?

Comment (scefali) [reply static/app/views/settings/organizationIntegrations/configureIntegration.tsx:170]: Pretty sure it's gonna say `jira_server` for the key

**Comment by ceorourke** [comment]
PR looks good but since you asked me about it earlier, what happens if the tasks fails / retries? Do we not care?

**Review by ceorourke — APPROVED** [review]

**Pull request merged**